cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/AbstractDataprocProvisioner.java

```diff
@@ -111,10 +112,19 @@ public final ClusterStatus deleteClusterWithStatus(ProvisionerContext context, C
     // Also cleanup files created by the job run.
     if (jobManager != null) {
       try {
-        RuntimeJobDetail jobDetail = jobManager.getDetail(context.getProgramRunInfo()).orElse(null);
+        DataprocRuntimeJobDetail jobDetail = (DataprocRuntimeJobDetail) jobManager.getDetail(
```

Comment:
It's safer to do a `instanceof` check before casting. So it's better be:

```java
RuntimeJobDetail jobDetail = jobManager.getDetail(...).orElse(null);
...
if (jobDetail != null && jobDetail.getStatus() == FAILED && (jobDetail instanceof DataprocRuntimeJobDetail)) {
  String statusDetails = ((DataprocRuntimeJobDetail) jobDetail).getJobStatusDetails();
  if (statusDetails != null) {
    LOG...
  }
}

Reply:
Done